apps/researcher/src/app/[locale]/objects/[id]/(provenance)/add-form.tsx

```diff
@@ -0,0 +1,455 @@
+import classNames from 'classnames';
+import {
+  LocalizedMarkdown,
+  useNotifications,
+  useSlideOut,
+} from '@colonial-collections/ui';
+import {
+  useForm,
+  SubmitHandler,
+  FormProvider,
+  useFormContext,
+} from 'react-hook-form';
+import {zodResolver} from '@hookform/resolvers/zod';
+import {z} from 'zod';
+import {useUser} from '@clerk/nextjs';
+import {addAttributionId} from '@/lib/user/actions';
+import edtf from 'edtf';
+import {
+  ButtonGroup,
+  FormWrapper,
+  FormColumn,
+  InputLabel,
+  SearchSelector,
+  LanguageSelector,
+  Textarea,
+  Input,
+  EdtfInput,
+  Select,
+  FieldValidationMessage,
+  isEdtf,
+} from '@/components/form';
+import {DefaultButton, PrimaryButton} from '@/components/buttons';
+import {ExclamationTriangleIcon} from '@heroicons/react/24/outline';
+import {CheckboxWithLabel} from '@/components/form/checkbox-with-label';
+import {addProvenanceEnrichment} from './actions';
+import {UserTypeOption, typeMapping} from './type-mapping';
+
+interface FormValues {
+  attributionId: string;
+  citation: string;
+  inLanguage?: string;
+  transferredFrom: {id: string; name: string};
+  transferredTo: {id: string; name: string};
+  location: {id: string; name: string};
+  type: {id: string; name: string};
+  date: {
+    startDate: string;
+    endDate: string;
+  };
+  agreedToLicense: boolean;
+}
+
+interface Props {
+  objectId: string;
+  slideOutId: string;
+}
+
+export default function AddProvenanceForm({objectId, slideOutId}: Props) {
+  const t = useTranslations('ProvenanceForm');
+  const tType = useTranslations('ProvenanceEventType');
+
+  const locale = useLocale();
+  const {user} = useUser();
+
+  const attributionIds = useMemo(
+    () => user?.publicMetadata?.attributionIds as string[] | undefined,
+    [user?.publicMetadata?.attributionIds]
+  );
+  const {addNotification} = useNotifications();
+  const {setIsVisible} = useSlideOut();
+
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  const provenanceEnricherSchema = z.object({
+    citation: z
+      .string()
+      .trim()
+      .min(1, {message: t('citationRequired')}),
+    inLanguage: z.string().optional(),
+    attributionId: z.string().url({message: t('invalidAttributionId')}),
+    agreedToLicense: z.literal<boolean>(true, {
+      errorMap: () => ({message: t('agreedToLicenseUnchecked')}),
+    }),
+    type: z.object({
+      id: z.nativeEnum(UserTypeOption, {
+        errorMap: () => ({message: t('typeRequired')}),
+      }),
+      name: z.string(),
+    }),
+    date: z
+      .object({
+        startDate: z.union([
+          z.literal(''),
+          z.string().refine(isEdtf, {message: t('invalidStartDate')}),
+        ]),
+        endDate: z.union([
+          z.literal(''),
+          z.string().refine(isEdtf, {message: t('invalidEndDate')}),
+        ]),
+      })
+      .refine(
+        ({startDate, endDate}) => {
+          if (!isEdtf(startDate) || !isEdtf(endDate)) return true;
```

Comment:
To be sure: is the OR condition correct (it's not an AND)?

Reply:
Yes, if one is not a valid date, then there is no range. So a range validation is not needed.

A not-valid edtf can mean that it is empty (which is valid) or has a weird value, in which case the validation will be done on the start/end date level.